zerver/decorator.py

```diff
@@ -735,7 +735,7 @@ def wrapped_func(*args, **kwargs):
     return wrapped_func  # type: ignore # https://github.com/python/mypy/issues/1927
 
 def return_success_on_head_request(view_func):
-    # type: (Callable) -> Callable
+    # type: (Callable) -> Union[Callable, HttpResponse]
```

Comment:
This change looks wrong to me; I think maybe you mean to change `_wrapped_view_func` below to return an HttpResponse (unconditionally)?